src/Canvas.cc

```diff
@@ -50,25 +50,25 @@ Canvas::Initialize(Napi::Env& env, Napi::Object& exports) {
 
   // Constructor
   Napi::Function ctor = DefineClass(env, "Canvas", {
-    InstanceMethod<&Canvas::ToBuffer>("toBuffer"),
-    InstanceMethod<&Canvas::StreamPNGSync>("streamPNGSync"),
-    InstanceMethod<&Canvas::StreamPDFSync>("streamPDFSync"),
+    InstanceMethod<&Canvas::ToBuffer>("toBuffer", napi_default_method),
+    InstanceMethod<&Canvas::StreamPNGSync>("streamPNGSync", napi_default_method),
+    InstanceMethod<&Canvas::StreamPDFSync>("streamPDFSync", napi_default_method),
 #ifdef HAVE_JPEG
-    InstanceMethod<&Canvas::StreamJPEGSync>("streamJPEGSync"),
+    InstanceMethod<&Canvas::StreamJPEGSync>("streamJPEGSync", napi_default_method),
 #endif
-    InstanceAccessor<&Canvas::GetType>("type"),
-    InstanceAccessor<&Canvas::GetStride>("stride"),
-    InstanceAccessor<&Canvas::GetWidth, &Canvas::SetWidth>("width"),
-    InstanceAccessor<&Canvas::GetHeight, &Canvas::SetHeight>("height"),
-    InstanceValue("PNG_NO_FILTERS", Napi::Number::New(env, PNG_NO_FILTERS)),
-    InstanceValue("PNG_FILTER_NONE", Napi::Number::New(env, PNG_FILTER_NONE)),
-    InstanceValue("PNG_FILTER_SUB", Napi::Number::New(env, PNG_FILTER_SUB)),
-    InstanceValue("PNG_FILTER_UP", Napi::Number::New(env, PNG_FILTER_UP)),
-    InstanceValue("PNG_FILTER_AVG", Napi::Number::New(env, PNG_FILTER_AVG)),
-    InstanceValue("PNG_FILTER_PAETH", Napi::Number::New(env, PNG_FILTER_PAETH)),
-    InstanceValue("PNG_ALL_FILTERS", Napi::Number::New(env, PNG_ALL_FILTERS)),
-    StaticMethod<&Canvas::RegisterFont>("_registerFont"),
-    StaticMethod<&Canvas::DeregisterAllFonts>("_deregisterAllFonts")
+    InstanceAccessor<&Canvas::GetType>("type", napi_default_jsproperty),
+    InstanceAccessor<&Canvas::GetStride>("stride", napi_default_jsproperty),
+    InstanceAccessor<&Canvas::GetWidth, &Canvas::SetWidth>("width", napi_default_jsproperty),
+    InstanceAccessor<&Canvas::GetHeight, &Canvas::SetHeight>("height", napi_default_jsproperty),
+    StaticValue("PNG_NO_FILTERS", Napi::Number::New(env, PNG_NO_FILTERS), napi_default_jsproperty),
```

Comment:
Changing these to a static value is a breaking change. This goes all the way back to 2013: 14d9f587b6106dd900cabe0d6bc31117acb596c6.

I think it's more correct to put on the class, but it definitely needs to be a documented change for 3.0.0, and the types need to be updated accordingly.

Reply:
I've added the types for this and a test that uses these